lib/rubocop/cop/style/endless_method_operator_precedence.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # Looks for endless methods inside operations of lower precedence (`and`, `or`, and
+      # modifier forms of `if`, `unless`, `while`, `until`) that are ambiguous due to
+      # lack of parentheses. This may lead to unexpected behavior as the code may appear
+      # to use these keywords as part of the method but in fact they modify
+      # the method definition itself.
+      #
+      # @safety
+      #   Autocorrection is unsafe as it changes potentially intended behavior; ie. if it
+      #   is actually desired to have the method definition modified by one of the keywords.
+      #   In this case, wrapping the method definition in parens will avoid the cop.
+      #
+      # @example
+      #
+      #   # bad
+      #   def foo = true and false
+      #
+      #   # good
+      #   def foo = (true and false)
+      #
+      #   # bad
+      #   def foo = true if bar
+      #
+      #   # good
+      #   def foo = (true if bar)
```

Comment:
I feel uncomfortable with presenting something that is not compatible with the bad example as good. This could be misleading for developers who properly understand the precedence of operators.

Reply:
Yes this is a good point that the good case has different behaviour than the bad case. I was thinking that the parens in the good case would be what was actually intended for the majority of cases like this, but making the parens explicit for the actual precedence is possible too. 

I am not sure that it is very likely that someone would write code along these lines and actually mean for the and/or/if etc. to modify the `def` rather than the `def` body, but I could be wrong. Maybe @phene or @vlad-pisanov can weigh in given their participation in the issue.

Reply:
It doesn't seem appropriate to use endless method definitions in cases where they can cause confusion. Methods that can lead to ambiguity without specifying operator precedence are likely not suited for endless method definitions in the first place.
In such cases, I think it's better to suggest using the regular method definition style. For example, the following would be bad and good examples:

```ruby
      #   # bad
      #   def foo = true and false
      #
      #   # good - When the method body is the intended focus. This is incompatible behavior.
      #   def foo
      #     true and false
      #   end
      #
      #   # good - When the intended meaning is clear. This is compatible behavior.
      #   def foo
      #     true
      #   end and false
```

Also, because the user intent can't be made completely clear, I think autocorrect should not be supported.